spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java

```diff
@@ -514,6 +517,20 @@ public void setSize(Integer size) {
 		@NestedConfigurationProperty
 		private final ListenerRetry retry = new ListenerRetry();
 
+		@NestedConfigurationProperty
+		private final SimpleContainer simple = new SimpleContainer();
```

Comment:
@snicoll The remaining properties (in `Listener`) are common to both containers (on the abstract container). The SMLC has 3 properties that are not available with the DMLC and it has one property that's not available. I don't mind having them at the top `Listener` level and simply ignoring those that don't apply, based on the `containerType`; it would certainly be better for backwards compatibility. I thought you preferred this hierarchical approach.